client-sdk/ts-web/core/src/signature.ts

```diff
@@ -168,10 +168,10 @@ export class NaclSigner implements Signer {
 }
 
 export type MessageHandlerBare<PARSED> = (v: PARSED) => void;
-export type MessageHandlersBare = {[context: string]: MessageHandlerBare<unknown>};
+export type MessageHandlersBare = {[context: string]: MessageHandlerBare<never>};
```

Comment:
using bottom type so that modules can conveniently provide contravariant handler of their specific body type. but then things get annoying when we call. lmk if you have better ideas for how to do this

Reply:
come to think of it, if there's going to be annoying casting on every dispatch, maybe we should simplify and use `{}`

Reply:
ok I made it even weirder by casting `fromCBOR(...) as never` instead of `handler as MessageHandlerBare<unknown>` :upside_down_face:

Reply:
how come `never` instead of `any`?

Reply:
the parameter type is already `never` :laughing:

Reply:
ok other answer now that I thought about it some more:

`never` makes sense here because statically we don't know of a type that will work as a parameter here. we couldn't, because it's up to the more specialized runtime modules to know what parameter types go with what contexts. if typescript is meant to answer the question "is it guaranteed to be safe if we call the handler with an argument of type X?" then for whatever type X we are able to name in this part of the code, the answer is, correctly, "no." and the contravariance works out too, with those more specialized contexts permitted to substitute a MessageHandlerBare<Something> where a MessageHandlerBare<never> is called for.

for practical purposes, `never` forces you to cast, which is more correct if you genuinely don't know what the parameter type is supposed to be. `any` saves that cast, but we don't particularly need it.

and we used to use `unknown`, which incorrectly meant that the handler should be able to handle anything and internally check the type